File: public/static/templates/spring-boot-clean/src/main/java/com/eventstorming/external/ExternalEntity.java
forEach: RelationCommandInfo
fileName: {{commandValue.aggregate.namePascalCase}}.java
path: {{boundedContext.name}}/{{{options.packagePath}}}/external
---
package {{options.package}}.external;

{{#commandValue.aggregate}}
public class {{namePascalCase}} {

    {{#aggregateRoot.fieldDescriptors}}
    private {{className}} {{nameCamelCase}};
    {{/aggregateRoot.fieldDescriptors}}

{{#aggregateRoot.fieldDescriptors}}
    public {{className}} get{{namePascalCase}}() {
        return {{nameCamelCase}};
    }
    public void set{{namePascalCase}}({{className}} {{nameCamelCase}}) {
        this.{{nameCamelCase}} = {{nameCamelCase}};
    }
{{/aggregateRoot.fieldDescriptors}}

}
{{/commandValue.aggregate}}

File: public/static/templates/nodejs/external/ExternalEntity.js
forEach: RelationCommandInfo
fileName: {{commandValue.aggregate.namePascalCase}}.js
path: {{boundedContext.name}}/external
---
{{#commandValue.aggregate}}
class {{namePascalCase}}{
    constructor(){
        {{#aggregateRoot.fieldDescriptors}}
        this.{{nameCamelCase}} = null;
        {{/aggregateRoot.fieldDescriptors}}
    }
}

module.exports = {{namePascalCase}};
{{/commandValue.aggregate}}

